integration-test/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/backend/minikube/MinikubeTestBackend.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.k8s.integrationtest.backend.minikube
+
+import io.fabric8.kubernetes.client.DefaultKubernetesClient
+
+import org.apache.spark.deploy.k8s.integrationtest.backend.IntegrationTestBackend
+import org.apache.spark.deploy.k8s.integrationtest.constants.MINIKUBE_TEST_BACKEND
+import org.apache.spark.deploy.k8s.integrationtest.docker.SparkDockerImageBuilder
```

Comment:
I believe this is using the incorrect branch - shouldn't this be based on https://github.com/apache-spark-on-k8s/spark/pull/521 which in turn assumes that Minikube is pre-installed on the box? The code here should reflect what's in the `integration-tests-reuse-minikube` branch in apache-spark-on-k8s.

Reply:
Yeah, I tried to start with apache-spark-on-k8s/spark#521. But I had to give up after I realized there are too many things to do even with what's already in the fork. Mostly decoupling the maven dependencies took me 2 full days including Saturday afternoon. A lot of fun :-)

I think we want to add #521 later in a follow-up PR. Maybe others can chip in once the basic overhaul is done by this PR.

Reply:
Hm, I'm not sure what's in #521 that's specific to the fork that isn't in this PR?

Reply:
Specifically, looking at the diff in the PR, it's entirely code level. There aren't any new Maven constructs added or taken away. So we should be able to apply the diff here as well without needing to worry about Maven at all?

Reply:
I am not familiar with #521 yet. So I don't know how much extra time it would need to address the code and potential review comments that the new code would invite. And I personally don't have a lot of extra time to spend on this beyond what I spent already. 

I also don't know much about the riselab setup. I was hoping to delay that until @ssuchter comes back. We can use Pepperdata jenkins in the meantime.

So I really hope that we can keep this PR simple and address gaps in future PRs.

Reply:
Followed up offline - we can introduce the usage of a pre-installed Minikube later, and in fact we can't use a pre-installed Minikube right now since our first iteration of this will be deploying the tests on Pepperdata's Jenkins nodes. The jenkins jobs will be serialized with builds from apache-spark-on-k8s, so we shouldn't have multiple test runs colliding on managing the Minikube VM.

There needs to be a specific sequencing of events to switch to using a pre-installed Minikube instance on the Riselab jenkins nodes instead. I'll follow up on what that specific sequencing is in an issue.

Reply:
Filed https://github.com/apache-spark-on-k8s/spark-integration/issues/4 to track.